app/src/main/java/org/sopt/dosopttemplate/Home/HomeActivity.kt

```diff
@@ -0,0 +1,170 @@
+        super.onCreate(savedInstanceState)
+        binding = ActivityHomeBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val currentFragment = supportFragmentManager.findFragmentById(R.id.fcv_home)
+        if (currentFragment == null) {
+            supportFragmentManager.beginTransaction()
+                .add(R.id.fcv_home, HomeFragment())
+                .commit()
+        }
+        clickBottomNavigation()
+        setCLICKFAB()
+        //checkManageExternalStoragePermission()
+    }
+
+    companion object {
+        fun createMyPageFragment(user_id: String?, user_major: String?): MyPageFragment {
+            return if (user_id != null && user_major != null) {
+                MyPageFragment.newInstance(user_id, user_major)
+            } else {
+                throw IllegalArgumentException("데이터가 없어요!")
+            }
+        }
+    }
+
+    private fun clickBottomNavigation() {
+        binding.bnvHome.setOnItemSelectedListener {
+            when (it.itemId) {
+                R.id.menu_home -> {
+                    replaceFragment(UserFragment())
+                    true
+                }
+
+                R.id.menu_do_android -> {
+                    replaceFragment(DoAndroidFragment())
+                    true
+                }
+
+                R.id.menu_mypage -> {
+                    try {
+                        val user_id = intent?.getStringExtra("user_id")
+                        val user_major = intent?.getStringExtra("user_major")
+                        val myPageFragment = createMyPageFragment(user_id, user_major)
+                        replaceFragment(myPageFragment)
+                    } catch (e: IllegalArgumentException) {
+                        Toast.makeText(this, e.message, Toast.LENGTH_SHORT).show()
+                    }
+                    true
+                }
+
+                else -> false
+            }
+        }
+    }
+
+    private fun replaceFragment(fragment: Fragment) {
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.fcv_home, fragment)
+            .commit()
+
+    }
+
+    private fun setCLICKFAB() {
+        binding.fabMain.setOnClickListener {
+            eventFAB()
+        }
+        binding.fabCapture.setOnClickListener {
+            //takeScreenshot()
+        }
+
+        binding.fabShare.setOnClickListener {
+            Toast.makeText(this, "공유하는 중", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun eventFAB() {
+        if (openFAB) {
+            ObjectAnimator.ofFloat(binding.fabShare, "translationY", 0f).apply { start() }
+            ObjectAnimator.ofFloat(binding.fabCapture, "translationY", 0f).apply { start() }
+        } else {
+            ObjectAnimator.ofFloat(binding.fabShare, "translationY", -400f).apply { start() }
+            ObjectAnimator.ofFloat(binding.fabCapture, "translationY", -200f).apply { start() }
+        }
```

Comment:
대부분의 문장이 반복되는 것 같습니다! 바뀌는 부분만 변수처리해도 좋을 것 같네용
```suggestion      
        var fir = 0f // 무슨 기능인지 잘 몰라서 변수명은 대충지었습니다!
        var sec = 0f
        
        if (openFAB) {
            fir = -400f
            sec = -200f
        }
        
        ObjectAnimator.ofFloat(binding.fabShare, "translationY", fir).apply { start() }
        ObjectAnimator.ofFloat(binding.fabCapture, "translationY", sec).apply { start() }
```